src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs

```diff
@@ -251,5 +251,152 @@ public TestRunFullResults RunTests()
+                    else
+                    {
+                        if (row.IsValue)
+                        {
+                            TestToString(row.Value, sb);
+                        }
+                        else
+                        {
+                            TestToString(row.ToFormulaValue(), sb);
+                        }
+                    }
+
+                    dil = ",";
+                }
+
+                if (canUseSquareBracketSyntax)
+                {
+                    sb.Append(']');
+                }
+                else
+                {
+                    sb.Append(')');
+                }
+            }
+            else if (result is RecordValue r)
+            {
+                var fields = r.Fields.ToArray();
+                Array.Sort(fields, (a, b) => string.CompareOrdinal(a.Name, b.Name));
+
+                sb.Append('{');
+                var dil = string.Empty;
+
+                foreach (var field in fields)
+                {
+                    sb.Append(dil);
+                    sb.Append(field.Name);
+                    sb.Append(':');
+                    TestToString(field.Value, sb);
+
+                    dil = ",";
+                }
+
+                sb.Append('}');
+            }
+            else if (result is TimeValue tv)
+            {
+                // Time(HH,mm,ss,fff)
+                var timeSpan = tv.Value;
+                sb.Append($"Time({timeSpan.Hours},{timeSpan.Minutes},{timeSpan.Seconds},{timeSpan.Milliseconds})");
+            }
+            else if (result is BlankValue)
+            {
+                sb.Append("Blank()");
+            }
+            else if (result is DateValue d)
+            {
+                // Date(YYYY,MM,DD)
+                var date = d.Value;
+                sb.Append($"Date({date.Year},{date.Month},{date.Day})");
+            }
+            else if (result is DateTimeValue dt)
+            {
+                // DateTime(yyyy,MM,dd,HH,mm,ss,fff)
+                var dateTime = dt.Value;
+                sb.Append($"DateTime({dateTime.Year},{dateTime.Month},{dateTime.Day},{dateTime.Hour},{dateTime.Minute},{dateTime.Second},{dateTime.Millisecond})");
+            }
+            else if (result is OptionSetValue opt)
+            {
+                sb.Append($"{opt.Type.OptionSetName}.{opt.Option}");
+            }
+            else if (result is GuidValue gv)
+            {
+                sb.Append($"GUID(\"{gv.Value}\")");
+            }
+            else if (result is ColorValue cv)
+            {   
+                sb.Append($"RGBA({cv.Value.R},{cv.Value.G},{cv.Value.B},{Math.Round(cv.Value.A / 255.0, 3)})");
+            }
+            else if (result is ErrorValue)
+            {
+                sb.Append(result);
+            }
+            else
+            {
+                throw new InvalidOperationException($"unsupported value type: {result.GetType().Name}");
+            }
+        }
```

Comment:
We don't use this TestToString anymore; it's now part of the serialization.

Reply:
updated.